redis_hashring/__init__.py

```diff
@@ -320,12 +317,12 @@ def poll(self, as_greenlet=False):
 
         try:
             while True:
+                # Since Redis' listen method blocks, we use select to inspect the
+                # underlying socket to see if there is activity.
+                fileno = pubsub.connection._sock.fileno()
```

Comment:
So `_sock` is set because we `subscribe()` above. Is there any scenario where `_sock` would be set back to `None` we should be aware of? If we haven't run into any issues before then it's probably fine.

Reply:
Not that I can see. From inside the PubSub class itself, `self.connection = None` only when you call `self.reset()`, and that is called when using PubSub as a context manager, when the object is being garbage collected (via `__del__`) and when you call `self.close()`. We never call `pubsub.reset()` from our code, and we call `pubsub.close()` only when `self.poll` ends. I believe we are good.